editor/code_editor.cpp

```diff
@@ -142,22 +142,26 @@ void FindReplaceBar::unhandled_input(const Ref<InputEvent> &p_event) {
 
 bool FindReplaceBar::_search(uint32_t p_flags, int p_from_line, int p_from_col) {
 	String text = get_search_text();
-	Point2i pos = text_editor->search(text, p_flags, p_from_line, p_from_col);
 
-	if (pos.x != -1) {
+	if (search_result)
+		memdelete(search_result);
+
+	search_result = text_editor->search(text, p_flags, p_from_line, p_from_col);
```

Comment:
Search results gonna leak. It's allocated, but not freed, unless you call `_search` again.